src/neural/cuda/network_cuda.cc

```diff
@@ -192,10 +192,13 @@ class CudaNetwork : public Network {
               "using a smaller network.";
     }
 
-    // Disable res block fusing for > 384 filters (the fused output input
+    // Disable res block fusing for > 512 filters (the fused output input
     // transform kernel runs out of register space) and for fp32 for now.
     // TODO: make it work for filters not a multiple of 32.
-    if (kNumFilters <= 384 && kNumFilters % 32 == 0 &&
+    if ((kNumFilters <= kMaxSupportedChannelsForResBlockFusing ||
+         (deviceProp.major >= 8 && 
```

Comment:
If I'm reading the specs right, V100 and Jetson Xavier can also take advantage of this, so this line could be something like:
```c++
((deviceProp.major >= 8 || (deviceProp.major == 7 && deviceProp.minor != 5)) &&
```

Reply:
Thanks Borg! You are right, Volta architecture should still support the optimization as it has 96KB of shared memory available. I have updated the check.